utils/network.go

```diff
@@ -340,3 +345,41 @@ func writeProcSys(path, value string) error {
 	}
 	return err
 }
+
+// Generate a mac address from the passed in string
+func genLocalMacAddr(src string) net.HardwareAddr {
```

Comment:
would be nice to see some UTs to ensure correct behavior (and resilience in the face of unexpected input) of `genLocalMacAddr`. Some nice test cases:
* length == 0
* invalid characters
* fewer characters (or non-zero characters) than are present in a mac address